File: May-LeetCoding-Challange-2020/Week1/771-jewels-and-stones.py
class Solution:
    # Sol 1. Hash Set
    def numJewelsInStones(self, J: str, S: str) -> int:
        jewels = set(J)
        how_many_jewels = 0
        for s in S:
            if s in jewels:
                how_many_jewels += 1
        return how_many_jewels
    # Time Complexity: O(n)
    # Space Complexity: O(n)

File: May-LeetCoding-Challange-2020/771-jewels_and_stones.cpp
class Solution {
public:

    /*
    *
    *  Time Complexity: O(n^2)
    *  Space Complexity: O(1)
    * 
    */
    int numJewelsInStones(string J, string S) {
        int count = 0;
        
        for(int i = 0; i < S.length(); i++){
            for(int j = 0; j < J.length(); j++){
                if(S[i] == J[j]){
                    count++;
                }
            }
        }
        
        return count;
    }
};